packages/server/lib/controllers/v1/flows/id/patchFrequency.ts

```diff
@@ -8,7 +8,13 @@ import { z } from 'zod';
 
 const orchestrator = getOrchestrator();
 export const validationBody = validationBodyBase.extend({
-    frequency: z.string().regex(/^(?<every>every )?(?<amount>[0-9])?\s?(?<unit>(m|mins?|minutes?|h|hrs?|hours?|d|days?))$/)
+    // To sync with ScriptSettings
+    // Test: https://regex101.com/r/gJBaKt
+    frequency: z
+        .string()
+        .regex(
+            /^(?<every>every )?((?<amount>[0-9]+)?\s?(?<unit>(m|mins?|minutes?|h|hrs?|hours?|d|days?))|(?<unit2>(month|week|half day|half hour|quarter hour)))$/
```

Comment:
instead of validating at the request level we could just get a string and let the frequency update fails if the value is not valid

Reply:
I am not a fan of letting an invalid value enter the system. We could share the frequency validation function across the services